Removing an import cause an NPE when the same package is reimported

**Description:**
Consider the following code:
```
import ballerina/http;

function myFunction() returns error? {
    http:<cursor>
}
```
At `<cursor>` we get suggestions as expected.
Now, remove the `ballerina/http` import:
```

function myFunction() returns error? {
    http:
}
```
And add `http` import again (so the code will be as in 1st scenario). Try completions (`ctrl+space`) at `cursor`. Following NPE is printed and completions do not work from thereafter for `http` module.
```java
[Error - 7:08:46 PM] Operation 'text/codeAction' failed! {uri: '/home/imesha/Documents/WSO2/Ballerina/Projects/test_project/test.bal', [1:1]- [2:1], error: 'failed to load the module 'ballerina/http:1.1.0-beta.2' from its BIR'}
org.ballerinalang.compiler.BLangCompilerException: failed to load the module 'ballerina/http:1.1.0-beta.2' from its BIR
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:191)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:167)
	at io.ballerina.projects.ModuleContext.loadPackageSymbolInternal(ModuleContext.java:443)
	at io.ballerina.projects.ModuleCompilationState$8.compile(ModuleCompilationState.java:232)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:329)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:174)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:166)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:95)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:206)
	at io.ballerina.projects.Package.getCompilation(Package.java:135)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.waitAndGetPackageCompilation(BallerinaWorkspaceManager.java:218)
	at org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager.semanticModel(BallerinaWorkspaceManager.java:197)
	at org.ballerinalang.langserver.codeaction.CodeActionRouter.getAvailableCodeActions(CodeActionRouter.java:66)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:46)
	at org.ballerinalang.langserver.codeaction.BallerinaCodeActionExtension.execute(BallerinaCodeActionExtension.java:34)
	at org.ballerinalang.langserver.LangExtensionDelegator.codeActions(LangExtensionDelegator.java:133)
	at org.ballerinalang.langserver.BallerinaTextDocumentService.lambda$codeAction$8(BallerinaTextDocumentService.java:289)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.NullPointerException
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver.lookupSymbol(SymbolResolver.java:546)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SymbolResolver.lookupSymbolInMainSpace(SymbolResolver.java:575)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readType(BIRPackageSymbolEnter.java:1113)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.readBType(BIRPackageSymbolEnter.java:578)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readTypeFromCp(BIRPackageSymbolEnter.java:986)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readType(BIRPackageSymbolEnter.java:1229)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.readBType(BIRPackageSymbolEnter.java:578)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readTypeFromCp(BIRPackageSymbolEnter.java:986)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readType(BIRPackageSymbolEnter.java:1271)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.readBType(BIRPackageSymbolEnter.java:578)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readTypeFromCp(BIRPackageSymbolEnter.java:986)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter$BIRTypeReader.readType(BIRPackageSymbolEnter.java:1067)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.readBType(BIRPackageSymbolEnter.java:578)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.defineTypeDef(BIRPackageSymbolEnter.java:465)
	at org.wso2.ballerinalang.util.LambdaExceptionUtils.lambda$rethrow$0(LambdaExceptionUtils.java:45)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.defineSymbols(BIRPackageSymbolEnter.java:360)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:237)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:213)
	at org.wso2.ballerinalang.compiler.BIRPackageSymbolEnter.definePackage(BIRPackageSymbolEnter.java:186)
	... 23 more
```
Seems like the dependencies of `http` module are now not in the package cache.

**Steps to reproduce:**
See description

**Affected Versions:**
SL Beta 2 SNAPSHOT

--- a/compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java
+++ b/compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java
@@ -430,13 +430,15 @@ public class Package {
                 PackageCache packageCache = PackageCache.getInstance(compilerContext);
                 for (ResolvedPackageDependency dependency : diff) {
                     for (ModuleId moduleId : dependency.packageInstance().moduleIds()) {
-                        Module module = dependency.packageInstance().module(moduleId);
-                        PackageID packageID = module.descriptor().moduleCompilationId();
-                        // remove the module from the compiler packageCache
-                        packageCache.remove(packageID);
-                        // we need to also reset the module in the project environment packageCache
-                        // to make the module recompile and add symbols
-                        module.modify().apply();
+                        if (!dependency.packageInstance().descriptor().isLangLibPackage()) {
+                            Module module = dependency.packageInstance().module(moduleId);
+                            PackageID packageID = module.descriptor().moduleCompilationId();
+                            // remove the module from the compiler packageCache
+                            packageCache.remove(packageID);
+                            // we need to also reset the module in the project environment packageCache
+                            // to make the module recompile and add symbols
+                            module.moduleContext().setCompilationState(null);
+                        }
                     }
                 }
             }

